src/processor.rs

```diff
@@ -39,18 +39,18 @@ pub async fn handle_request(
     payload: &mut web::Payload,
     req: &HttpRequest,
 ) -> HttpResponse {
-    let contents = match execute_function(function, payload, &headers, req).await {
+    let contents = match execute_function(function, payload, headers, req).await {
         Ok(res) => res,
         Err(err) => {
             println!("Error: {:?}", err);
             let mut response = HttpResponse::InternalServerError();
-            apply_headers(&mut response, &headers);
+            apply_headers(&mut response, headers);
             return response.finish();
         }
     };
 
     let mut response = HttpResponse::Ok();
-    apply_headers(&mut response, &headers);
+    apply_headers(&mut response, headers);
```

Comment:
Hi @SanchithHegde ,

Thank you for the PR. Why did you remove the `&` here?

Reply:
@sansyrox `clippy` throws this warning:

```
warning: this expression borrows a reference (`&std::sync::Arc<dashmap::DashMap<std::string::String, std::string::String
>>`) that is immediately dereferenced by the compiler
  --> src/processor.rs:53:34
   |
53 |     apply_headers(&mut response, &headers);
   |                                  ^^^^^^^^ help: change this to: `headers`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow
```

Reply:
@SanchithHegde , this makes sense. I will be reviewing all the PRs today. Will merge it then. Thanks again! 😄